File: problems/73/Main.py
#--SPLIT--

#--SPLIT--
# class Solution:
#     def primesum(self, a):

#--SPLIT--
class Solution:
    def primesum(self, a):
        isPrime = {}
        for i in range(2, a + 1):
            isPrime[i] = True

        for i in range(2, a + 1):
            for j in range(2, 1 + (a + 1) / i):
                isPrime[i * j] = False

        for i in range(2, a):
            j = a - i
            if (isPrime[i] and isPrime[j]):
                return [i, j]

        return None

#--SPLIT--

if __name__ == "__main__":
    test_cases = int(raw_input())
    for i in range(test_cases):
        a = int(raw_input())
        s = Solution()
        print(s.primesum(a))

File: problems/73/Main.c
#include <stdio.h>
#include <stdlib.h>

//--SPLIT--

/**
 * @input a : Integer
 *
 * @Output Integer array. You need to malloc memory, and fill the length in len1
 */
//-int* primesum(int a) {
//-}


//--SPLIT--
int * primesum(int a) {
    int result[2];
    int * isPrime = malloc(sizeof(int) * (a + 1));
    for (int i = 2; i <= a; i++) {
            isPrime[i] = 1;
    }
    for(int i = 2; i <=a; i++) {
        for (int j = 2; i * j <= a; j++) {
            isPrime[i * j] = 0;
        }
    }

    for (int i = 2; i < a; i++) {
        int j = a - i;
        if (isPrime[i] == 1 && isPrime[j] == 1) {
            result[0] = i;
            result[1] = j;
            return result;
        }
    }
    return result;
}

//--SPLIT--
int main() {
	setbuf(stdout, NULL);
	int testcases;
	scanf("%d", &testcases);
	int t = 0;
	for (t = 0; t < testcases; t++) {
		int a;
		scanf("%d", &a);
		int* result = primesum(a);
		printf("[%d, %d]\n", result[0], result[1]);
	}
	return 0;
}
